compiler/src/dmd/cppmangle.d

```diff
@@ -2522,58 +2522,70 @@ unittest
     assert(closestIndex([s1, s2, s4], s5, match) == 3 && !match);
 }
 
-/**
+/***
  * Visits the return type of a function and writes leftover ABI tags
+ * Params:
+ *   tf = Type of the function to mangle the return type of
+ *   previous = already written ones
+ *   toWrite = where to put StringExp's to be written
  */
-extern(C++) private final class LeftoverVisitor : Visitor
+private
+void leftOver(TypeFunction tf, const(Array!StringExp)* previous, Array!StringExp* toWrite)
 {
-    /// List of tags to write
-    private Array!StringExp toWrite;
-    /// List of tags to ignore
-    private const(Array!StringExp)* ignore;
-
-    ///
-    public this(const(Array!StringExp)* previous)
+    extern(C++) final class LeftoverVisitor : Visitor
     {
-        this.ignore = previous;
-    }
+        /// List of tags to write
+        private Array!StringExp* toWrite;
+        /// List of tags to ignore
+        private const(Array!StringExp)* ignore;
 
-    /// Reintroduce base class overloads
-    public alias visit = Visitor.visit;
+        ///
+        public this(const(Array!StringExp)* previous, Array!StringExp* toWrite)
+        {
+            this.ignore = previous;
+            this.toWrite = toWrite;
+        }
 
-    /// Least specialized overload of each direct child of `RootObject`
-    public override void visit(Dsymbol o)
-    {
-        auto ale = ABITagContainer.forSymbol(o);
-        if (!ale) return;
+        /// Reintroduce base class overloads
+        public alias visit = Visitor.visit;
 
-        bool match;
-        foreach (elem; *ale.elements)
+        /// Least specialized overload of each direct child of `RootObject`
+        public override void visit(Dsymbol o)
         {
-            auto se = elem.toStringExp();
-            closestIndex((*this.ignore)[], se, match);
-            if (match) continue;
-            auto idx = closestIndex(this.toWrite[], se, match);
-            if (!match)
-                this.toWrite.insert(idx, se);
+            auto ale = ABITagContainer.forSymbol(o);
+            if (!ale) return;
+
+            bool match;
+            foreach (elem; *ale.elements)
+            {
+                auto se = elem.toStringExp();
+                closestIndex((*this.ignore)[], se, match);
+                if (match) continue;
+                auto idx = closestIndex((*this.toWrite)[], se, match);
```

Comment:
Perhaps `toWrite` can be passed by `ref` so you don't need the parens and dereference.

Reply:
I'd eventually like to do that globally, but there are enough changes for this PR at the moment.